lib/libutee/tee_api_operations.c

```diff
@@ -329,35 +330,52 @@ TEE_Result TEE_SetOperationKey(TEE_OperationHandle operation,
 	if (key != TEE_HANDLE_NULL) {
 		TEE_ObjectInfo key_info;
 
-		TEE_GetObjectInfo(key, &key_info);
+		res = TEE_GetObjectInfo1(key, &key_info);
+		if (res != TEE_SUCCESS && res != TEE_ERROR_CORRUPT_OBJECT &&
+		    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
+			TEE_Panic(0);
+
+		if (res == TEE_ERROR_CORRUPT_OBJECT ||
+		    res == TEE_ERROR_STORAGE_NOT_AVAILABLE)
+			return res;
+
 		/* Supplied key has to meet required usage */
 		if ((key_info.objectUsage & operation->info.requiredKeyUsage) !=
 		    operation->info.requiredKeyUsage) {
 			TEE_Panic(0);
 		}
 
-		if (operation->info.maxKeySize < key_info.objectSize)
+		if (operation->info.maxKeySize < key_info.keySize)
 			TEE_Panic(0);
 
-		key_size = key_info.objectSize;
+		key_size = key_info.keySize;
 	}
 
 	TEE_ResetTransientObject(operation->key1);
 	operation->info.handleState &= ~TEE_HANDLE_FLAG_KEY_SET;
 
 	if (key != TEE_HANDLE_NULL) {
-		TEE_CopyObjectAttributes(operation->key1, key);
+		res = TEE_CopyObjectAttributes1(operation->key1, key);
+		if (res != TEE_SUCCESS && res != TEE_ERROR_CORRUPT_OBJECT &&
+		    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
+			TEE_Panic(0);
+
+		if (res == TEE_ERROR_CORRUPT_OBJECT ||
```

Comment:
If this test and action is done before the panic test at line 359 above, then the panic test can be simplified.